python/plugins/db_manager/dlg_sql_window.py

```diff
@@ -91,10 +91,8 @@ def __init__(self, iface, db, parent=None):
         self.updatePresetsCombobox()
 
         self.geomCombo.setEditable(True)
-        self.geomCombo.lineEdit().setReadOnly(True)
 
         self.uniqueCombo.setEditable(True)
-        self.uniqueCombo.lineEdit().setReadOnly(True)
```

Comment:
When the provider is not able to detect the name of the geometry / unique id column. This might be the case for virtual layers in rare circumstances, for example when the query contains more than one geometry column